### PR TITLE
Mat 7897 ordering dependencies

### PR DESCRIPTION
**Pull request by gregory-akins**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-7897](https://jira.cms.gov/browse/MAT-7897)
(Optional) Related Tickets:

### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/service/CqlParsingService.java [hunk 1]
```diff
@@ -316,7 +335,9 @@ private ElementLookup buildElementLookupForValueSet(CQLValueSet cqlValueSet, Str
   }
 
   private Set<CQLDefinition> buildCqlDefinitions(CQLTools cqlTools) {
-    return cqlTools.getDefinitionContents().stream().map(this::buildCqlDefinition).collect(toSet());
+    Set<DefinitionContent> definitionContents = cqlTools.getDefinitionContents();
+    Map<String, Set<String>> usedDefinitions = cqlTools.getUsedDefinitions();
```
Comment (adongare) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/CqlParsingService.java]: looks like unused? or is this something that populates data?

**Review by adongare — DISMISSED** [review]

File: src/main/java/gov/cms/mat/cql_elm_translation/service/CqlParsingService.java (hunk 1 above)
Comment (gregory-akins) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/CqlParsingService.java]: Good catch.. I must have had a copy/paste error

**Review by adongare — APPROVED** [review]
